flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InternalMiniClusterExtension.java

```diff
@@ -26,11 +26,16 @@
 
 import java.net.URI;
 
-/** An extension which starts a {@link MiniCluster} for testing purposes. */
-public class MiniClusterExtension implements CustomExtension {
+/**
+ * An extension which starts a {@link MiniCluster} for testing purposes.
+ *
+ * <p>This should only be used by tests within the flink-runtime module. Other modules should use
+ * {@code MiniClusterExtension} provided by flink-test-utils module.
+ */
+public class InternalMiniClusterExtension implements CustomExtension {
```

Comment:
Given that we ideally follow the existing pattern of the flink-test-utils version just being an extension of the flink-runtime version (see MiniClusterWithClientExtension/-Resource), this naming doesn't seem...ideal.

From a user-perspective it is certainly nicer if we'd just expose a `MiniClusterExtension` without anything else tacked on to it though.

Could this extension of the flink-runtime version also be achieved by wrapping it (in which case the naming would be fine-ish, although it'd still suck a bit in flink-runtime...)?

Reply:
> What do you mean by wrapping?

I mean that instead of wrapping a `MiniClusterResource` (which it currently does) it wraps a `InternalMiniClusterExtension`, and delegates some of the parameter injection to that wrapped extension (specifically, InjectMiniCluster/InjectClusterRESTAddress).

Reply:
Basically this: https://github.com/zentol/flink/commit/db42d836ed280c5c5789026e2b81612bd3506609

Reply:
https://github.com/zentol/flink/commit/c3ee7a74fb7c54e0f814e677b342c347c1a4325e